cpp/tests/text/normalize_tests.cpp

```diff
@@ -67,7 +67,59 @@ TEST_F(TextNormalizeTest, NormalizeEmptyTest)
   cudf::strings_column_view strings_view(strings->view());
-  auto const results = nvtext::normalize_spaces(strings_view);
+  auto results = nvtext::normalize_spaces(strings_view);
+  EXPECT_EQ(results->size(), 0);
+  results = nvtext::normalize_characters(strings_view, true);
+  EXPECT_EQ(results->size(), 0);
+  results = nvtext::normalize_characters(strings_view, false);
   EXPECT_EQ(results->size(), 0);
-  EXPECT_EQ(results->has_nulls(), false);
 }
+
+TEST_F(TextNormalizeTest, NormalizeCharacters)
+{
+  // These include punctuation, accents, whitespace, and CJK characters
+  std::vector<const char*> h_strings{"abc£def",
+                                     nullptr,
+                                     "éè â îô\taeio",
+                                     "\tĂĆĖÑ  Ü",
+                                     "ACEN U",
+                                     "P^NP",
+                                     "$41.07",
+                                     "[a,b]",
+                                     "丏丟",
+                                     ""};
+  auto validity =
+    thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; });
+  cudf::test::strings_column_wrapper strings(h_strings.begin(), h_strings.end(), validity);
+  cudf::strings_column_view strings_view(strings);
+  {
+    auto results = nvtext::normalize_characters(strings_view, true);
+    cudf::test::strings_column_wrapper expected({"abc£def",
+                                                 "",
+                                                 "ee a io aeio",
+                                                 " acen  u",
+                                                 "acen u",
+                                                 "p ^ np",
+                                                 " $ 41 . 07",
+                                                 " [ a , b ] ",
+                                                 " 丏  丟 ",
+                                                 ""},
+                                                validity);
+    cudf::test::expect_columns_equal(*results, expected);
+  }
+  {
+    auto results = nvtext::normalize_characters(cudf::strings_column_view(strings), false);
+    cudf::test::strings_column_wrapper expected({"abc£def",
+                                                 "",
+                                                 "éè â îô aeio",
+                                                 " ĂĆĖÑ  Ü",
+                                                 "ACEN U",
+                                                 "P ^ NP",
+                                                 " $ 41 . 07",
+                                                 " [ a , b ] ",
+                                                 " 丏  丟 ",
+                                                 ""},
+                                                validity);
+    cudf::test::expect_columns_equal(*results, expected);
+  }
+}
```

Comment:
I'm surprised this passes clang-format.

```suggestion

}
```